### PR TITLE
Add public examples to many functions to std.math

### PR DESCRIPTION
**Pull request by JackStouffer**

### CODE REVIEW

File: std/math.d [hunk 1]
```diff
@@ -3724,18 +3750,18 @@ real log10(real x) @safe pure nothrow @nogc
...
+ * log(1 + x).
  *
  *  $(TABLE_SV
  *  $(TR $(TH x)            $(TH log1p(x))     $(TH divide by 0?) $(TH invalid?))
  *  $(TR $(TD $(PLUSMN)0.0) $(TD $(PLUSMN)0.0) $(TD no)           $(TD no))
  *  $(TR $(TD -1.0)         $(TD -$(INFIN))    $(TD yes)          $(TD no))
- *  $(TR $(TD $(LT)-1.0)    $(TD $(NAN))       $(TD no)           $(TD yes))
- *  $(TR $(TD +$(INFIN))    $(TD -$(INFIN))    $(TD no)           $(TD no))
+ *  $(TR $(TD $(LT)-1.0)    $(TD -$(NAN))      $(TD no)           $(TD yes))
+ *  $(TR $(TD +$(INFIN))    $(TD +$(INFIN))    $(TD no)           $(TD no))
```
Comment (JackStouffer) [inline std/math.d]: these where incorrect

File: std/math.d [hunk 2]
```diff
@@ -4826,8 +4923,11 @@ version(Posix)
  * Returns the integer portion of x, dropping the fractional portion.
  *
  * This is also known as "chop" rounding.
+ *
+ * Returns:
+ *     A `real`.
  */
-real trunc(real x) @trusted nothrow @nogc
+auto trunc(real x) @trusted
```
Comment (JackStouffer) [inline std/math.d]: changed to `auto` to auto infer attributes like pure based on implementation without using template

File: std/math.d [hunk 3]
```diff
@@ -4870,6 +4970,15 @@ real trunc(real x) @trusted nothrow @nogc
...
 }
 
+///
+@safe pure unittest
+{
+    assert(trunc(0.01) == 0);
+    assert(trunc(0.49) == 0);
+    assert(trunc(0.5) == 0);
+    assert(trunc(1.5) == 1);
+}
```
Comment (wilzbach) [inline std/math.d]: Looks like the remainder isn't available on Win32?

```
std\math.d(4908): Error: static assert:  "lround not implemented"
```

File: std/math.d [hunk 4]
```diff
@@ -4809,12 +4905,13 @@ long lround(real x) @trusted nothrow @nogc
     version (Posix)
         return core.stdc.math.llroundl(x);
     else
-        assert(0, "lround not implemented");
+        static assert(0, "lround not implemented");
```
Comment (wilzbach) [inline std/math.d]: Looks like the remainder isn't available on Win32, but still run?

```
std\math.d(4908): Error: static assert:  "lround not implemented"
```

Maybe we should roll a custom, "slow" implementation of it?

File: std/math.d [hunk 5]
```diff
@@ -4536,24 +4585,46 @@ if (is(typeof(rfunc(F.init)) : F) && isFloatingPoint!F)
...
+ *
  * If the return value is not equal to x, the FE_INEXACT
  * exception is raised.
- * $(B nearbyint) performs
- * the same operation, but does not set the FE_INEXACT exception.
+ *
+ * $(LREF nearbyint) performs the same operation, but does
+ * not set the FE_INEXACT exception.
  */
 real rint(real x) @safe pure nothrow @nogc { pragma(inline, true); return core.math.rint(x); }
```
Comment (wilzbach) [inline std/math.d]: So we have `lround`, `nearbyint`, `round` and `rint` which all perform almost the same function. That's not very "abstracted high-level" over C  :/

File: std/math.d [hunk 6]
```diff
@@ -3724,18 +3750,18 @@ real log10(real x) @safe pure nothrow @nogc
...
+ * For very small x, log1p(x) will be more accurate than
+ * log(1 + x).
  *
  *  $(TABLE_SV
  *  $(TR $(TH x)            $(TH log1p(x))     $(TH divide by 0?) $(TH invalid?))
  *  $(TR $(TD $(PLUSMN)0.0) $(TD $(PLUSMN)0.0) $(TD no)           $(TD no))
  *  $(TR $(TD -1.0)         $(TD -$(INFIN))    $(TD yes)          $(TD no))
- *  $(TR $(TD $(LT)-1.0)    $(TD $(NAN))       $(TD no)           $(TD yes))
- *  $(TR $(TD +$(INFIN))    $(TD -$(INFIN))    $(TD no)           $(TD no))
+ *  $(TR $(TD $(LT)-1.0)    $(TD -$(NAN))      $(TD no)           $(TD yes))
```
Comment (wilzbach) [inline std/math.d]: Could you add an explicit test for this? (i.e. that it's a negative nan)

File: std/math.d (hunk 2 above)
Comment (n8sh) [reply std/math.d]: I feel the downside of not documenting the return type outweighs the benefit of attribute inference here.

Comment (JackStouffer) [reply std/math.d]: I added a note in the docs about the return type

File: std/math.d (hunk 6 above)
Comment (JackStouffer) [reply std/math.d]: I have a test for this on line 3798, but because `-real.nan != -real.nan` there's really no way to test this

Comment (n8sh) [reply std/math.d]: Use `is`.

File: std/math.d [hunk 7]
```diff
@@ -4812,62 +4919,88 @@ long lround(real x) @trusted nothrow @nogc
...
     }
 }
 
-/****************************************************
- * Returns the integer portion of x, dropping the fractional portion.
- *
- * This is also known as "chop" rounding.
- */
-real trunc(real x) @trusted nothrow @nogc
+version(StdDdoc)
```
Comment (JackStouffer) [inline std/math.d]: Ok, so apparently `ret` in ASM doesn't work with an `auto` return type. So we now have this mess.

**Pull request merged**